collective/googlenews/__init__.py

```diff
@@ -1,7 +1,40 @@
 # -*- coding: utf-8 -*-
-
 from zope.i18nmessageid import MessageFactory
-
 from collective.googlenews.config import PROJECTNAME
 
 _ = MessageFactory(PROJECTNAME)
+
+
+from collective.googlenews.interfaces import IGoogleNewsEditorPicksSettings
+from plone import api
+from plone.formwidget.namedfile.converter import b64decode_file
+from Products.CMFPlone.browser.syndication.adapters import FolderFeed
+
+import logging
+
+
+logger = logging.getLogger(PROJECTNAME)
+
+
+@property
+def logo(self):
+    """Return googlenews logo url if available, else return
+    portal logo url."""
+    logo = api.portal.get_registry_record(
+        IGoogleNewsEditorPicksSettings.__identifier__ +
+        '.googlenews_editorpicks'
+    )
+    if logo is None:
+        # return portal logo
+        return self._logo
+    # return google news logo
+    filename, data = b64decode_file(logo)
+    return '{0}/@@googlenews-editorpicks/{1}'.format(
+        api.portal.get().absolute_url(),
+        filename
+    )
+    return '%s/logo.png' % self.site.absolute_url()
+
+setattr(FolderFeed, '_logo', FolderFeed.logo)
```

Comment:
seems the canonical way of patching is stuff is by using collective.monkeypatcher:

http://docs.plone.org/4/en/develop/plone/misc/monkeypatch.html